src/_includes/async-await-2.0.md

```diff
@@ -1,6 +1,6 @@
 <aside class="alert alert-info" markdown="1">
-  **Dart 2 note:** In Dart 2, instead of immediately suspending, async functions
-  execute synchronously until the first await. In most cases, you won't notice a
-  difference in behavior relative to Dart 1&mdash;as long as you await each time you
-  call an async method, it won't be an issue.
+  **Version note:** In Dart 1.x, async functions immediately suspended
+  execution. In Dart 2, instead of immediately suspending, async functions
+  execute synchronously until the first `await`. If you use `await` each time
```

Comment:
oh man, I only just saw this.

Looks like my revisions here were maybe mostly unwarranted :(

I had this wrong:

```
Future<void> f() async {
  print('before await');
  await null;
  print('after await');
}

Future<void> wihoutAwait() async {
  print('before f()');
  f();
  print('after f()');
}

Future<void> withAwait() async {
  print('before await f()');
  await f();
  print('after await f()');
}
```

I thought Dart 2 changed the order of both `withAwait()` and `withoutAwait()`! But its only `withoutAwait()` that changed........and I changed code describing code that's more like `withAwait()`!

Reply:
Hmm, looks like maybe this section is wrong, or poorly worded?

"if your methods all begin with an await" its the same dart 2/ dart 1?

Reply:
I think that if all async methods begin with await, then the dart 1 & dart 2 behavior is the same. But since you might not control the implementation of the async methods you call, I think this note is concentrating on what you can do as a caller: don't count on the function returning immediately.

Reply:
Ah, so this is suggesting:

```
await null; // suspend
foo(); // async function
```

This could be clearer, and also seems unreasonable advise to follow. But its not wrong, so that's good.